libraries/GCS_MAVLink/GCS_Common.cpp

```diff
@@ -2960,43 +3384,66 @@ void GCS_MAVLINK::data_stream_send(void)
         DataFlash_Class::instance()->handle_log_send();
     }
 
-    gcs().set_out_of_time(false);
-
-    send_queued_parameters();
-
-    if (gcs().out_of_time()) return;
+    // push the queue for as long as we can:
+    retry_deferred();
+}
 
-    if (hal.scheduler->in_delay_callback()) {
-        if (in_hil_mode()) {
-            // in HIL we need to keep sending servo values to ensure
-            // the simulator doesn't pause, otherwise our sensor
-            // calibration could stall
-            if (stream_trigger(STREAM_RAW_CONTROLLER)) {
-                send_message(MSG_SERVO_OUT);
-            }
-            if (stream_trigger(STREAM_RC_CHANNELS)) {
-                send_message(MSG_SERVO_OUTPUT_RAW);
-            }
-        }
-        // send no other streams while in delay, just in case they
-        // take way too long to run
-        return;
+uint16_t GCS_MAVLINK::interval_for_stream(GCS_MAVLINK::streams id) const
+{
+    const float frate = streamRates[id].get();
+    if ((uint8_t)frate == 0) {
```

Comment:
is_zero ?

Reply:
No, we don't want to use that for integers.